llvm/test/CodeGen/ARM/scmp.ll

```diff
@@ -0,0 +1,155 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 5
+; RUN: llc -mtriple=thumbv7-apple-darwin %s -o - | FileCheck %s
+
+define i8 @scmp_8_8(i8 %x, i8 %y) nounwind {
```

Comment:
```suggestion
define i8 @scmp_8_8(i8 signext %x, i8 signext %y) nounwind {
```
And zeroext for the ucmp case. ARM ABI zero/sign extends small integers.

Reply:
Thanks, fixed it for all targets